src/core/composer/qgscomposertablev2.cpp

```diff
@@ -534,15 +541,15 @@ void QgsComposerTableV2::render( QPainter *p, const QRectF &, const int frameInd
     gridPen.setColor( mGridColor );
     gridPen.setJoinStyle( Qt::MiterJoin );
     p->setPen( gridPen );
-    drawHorizontalGridLines( p, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader );
-    drawVerticalGridLines( p, mMaxColumnWidthMap, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader, mergeCells );
+    if ( mHorizontalGrid ) drawHorizontalGridLines( p, rowsToShow.first, rowsToShow.second + numberEmptyRows, drawHeader );
```

Comment:
@nyalldawson , darn, I was sure I'd found the one scenario where single line if statement would be embraced ... ;)